python/paddle/fluid/framework.py

```diff
@@ -4148,6 +4164,91 @@ def __init__(self):
         # compiled program, i.e. Graph
         self._graph = None
 
+    def _find_var_class_kwargs(self, new_desc):
+        old_desc = self.desc
+        all_new_vars = []
+        block_num = new_desc.num_blocks()
+        for idx in range(block_num):
+            new_block_desc = new_desc.block(idx)
+            all_new_vars.append([])
+            block_new_vars = all_new_vars[-1]
+            for new_var_desc in new_block_desc.all_vars():
+                if self.blocks[idx].has_var(new_var_desc.name()):
```

Comment:
The block number of new_desc and old_desc is always equal?

Reply:
For now, it is true. We can fix this if this is not in the future.